crates/ruff/src/rules/flake8_simplify/rules/ast_if.rs

```diff
@@ -508,10 +511,11 @@ pub(crate) fn use_ternary_operator(checker: &mut Checker, stmt: &Stmt, parent: O
 
     // Don't flag if the resulting expression would exceed the maximum line length.
     let line_start = checker.locator.line_start(stmt.start());
-    if checker.locator.contents()[TextRange::new(line_start, stmt.start())].width()
-        + contents.width()
-        > checker.settings.line_length
-    {
+    let tab_size = checker.settings.tab_size;
+    let mut width = checker.locator.contents()[TextRange::new(line_start, stmt.start())]
+        .width_with_tabs(tab_size, None);
+    width = contents.width_with_tabs(tab_size, Some(width));
```

Comment:
Hmm this is interesting. I wouldn't have thought that I need to pass the column when writing this code myself. I wonder if we could design the API in a way to guide users to do the right "thing" by default.

Maybe

```
LineWidth::from_start_of_line(text, tab_width) -> LineWidth
LineWidth::add_str(self, text) -> LineWidth
```

Using a newtype wrapper makes it a bit harder to get this wrong because it e.g. doesn't support adding to line widths with `+`